atomic_reactor/util.py

```diff
@@ -234,41 +235,40 @@ def wait_for_command(logs_generator):
     return cr
 
 
-def clone_git_repo(git_url, target_dir, commit=None):
+def clone_git_repo(git_url, target_dir, commit=None, retry_times=GIT_MAX_RETRIES):
     """
     clone provided git repo to target_dir, optionally checkout provided commit
 
     :param git_url: str, git repo to clone
     :param target_dir: str, filesystem path where the repo should be cloned
     :param commit: str, commit to checkout, SHA-1 or ref
+    :param retry_times: int, number of retries for git clone
     :return: str, commit ID of HEAD
     """
+    retry_delay = GIT_BACKOFF_FACTOR
+
     commit = commit or "master"
     logger.info("cloning git repo '%s'", git_url)
     logger.debug("url = '%s', dir = '%s', commit = '%s'",
                  git_url, target_dir, commit)
 
-    cmd = ["git", "clone", "-b", commit, "--depth", "1", git_url, quote(target_dir)]
-    logger.debug("doing a shallow clone '%s'", cmd)
-    try:
-        subprocess.check_call(cmd)
-    except subprocess.CalledProcessError as ex:
-        logger.warning(repr(ex))
-        # http://stackoverflow.com/questions/1911109/clone-a-specific-git-branch/4568323#4568323
-        # -b takes only refs, not SHA-1
-        cmd = ["git", "clone", "-b", commit, "--single-branch", git_url, quote(target_dir)]
-        logger.debug("cloning single branch '%s'", cmd)
+    cmd = ["git", "clone", git_url, quote(target_dir)]
+
+    for counter in range(retry_times + 1):
         try:
-            subprocess.check_call(cmd)
-        except subprocess.CalledProcessError as ex:
-            logger.warning(repr(ex))
-            # let's try again with plain `git clone $url && git checkout`
-            cmd = ["git", "clone", git_url, quote(target_dir)]
             logger.debug("cloning '%s'", cmd)
```

Comment:
Nitpick: move debug call out of the loop so it won't be repeated on retries

Reply:
ok